src/app/check-auth.js

```diff
@@ -23,27 +24,56 @@ const actionWhitelist = [
 // craco.config.js so that the name doesn't get mangled by CRA.
 const thunkWhitelist = ["connectAndStartPolling", "logOut"];
 
-function error(name) {
-  console.log("unable to perform action:", name, "user not authenticated");
+function error(name, wsControllerURL) {
+  console.log(
+    "unable to perform action:",
+    name,
+    "user not authenticated for:",
+    wsControllerURL
+  );
 }
 
-export default ({ getState }) => (next) => async (action) => {
-  const loggedIn = isLoggedIn(getState());
+const checkLoggedIn = (state, wsControllerURL) => {
+  if (!wsControllerURL) {
+    console.error("unable to determine logged in status");
+  }
+  return isLoggedIn(wsControllerURL, state);
+};
+
+/**
+  Redux middleware to enable gating actions on the respective controller
+  authentication.
+  @param {Object} action The typical Redux action or thunk to execute
+  @param {Object} options Any options that this checker needs to perform an
+    appropriate auth check.
+      wsControllerURL: The full controller websocket url that the controller
+        is stored under in redux in order to determine it's logged in status.
+*/
+export default ({ getState }) => (next) => async (action, options) => {
```

Comment:
This change adds an additional argument to the `dispatch` method. It's call signature now requires the `wsControllerURL` to perform actions that involve a specific controller.
```js
dispatch(action, options)
```